torch/csrc/Exceptions.h

```diff
@@ -92,6 +92,11 @@ static inline void PyErr_SetString(PyObject* type, const std::string& message) {
     PyErr_SetString(THPException_LinAlgError, torch::processErrorMsg(msg));  \
     retstmnt;                                                                \
   }                                                                          \
+  catch (const c10::CUDAOutOfMemoryError& e) {                               \
+    auto msg = e.what();                                                     \
+    PyErr_SetString(THPException_LinAlgError, torch::processErrorMsg(msg));  \
+    retstmnt;                                                                \
+  }                                                                          \
   catch (const c10::Error& e) {                                              \
     auto msg = torch::get_cpp_stacktraces_enabled()                          \
```

Comment:
Maybe we should just add a bool to c10::Error to indicate whether an error type should always include the stack and just put the logic here. CUDAOutOfMemoryError could then set the field to true in its constructor.

Reply:
Filed https://github.com/pytorch/pytorch/issues/82751 to track this as follow-up work.